espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Router.java

```diff
@@ -45,69 +48,81 @@ private void addRoute(Method method, String uri, RequestHandler handler) throws
+                return new NotFoundResponse();
+            }
+        };
+
+        Map<String, Object> params = parseBody(session);
+
+        // Look for a matching route
+        // TODO: Move this to a separate method 'isRouteMatch'.
+        for (Map.Entry<String, RequestHandler> entry : routerMap.get(method).entrySet()) {
+            String testUri = entry.getKey();
+
+            // TODO: Use StringBuilder to construct the Test Regexes
+            String testRegex = "^";
+            Map<Integer, String> wildcardIndices = new HashMap<>();
+
+            // Convert route to a regex to test incoming URI against
+            // TODO: Cache these regexes instead of re-creating them every time
+            int index = 0;
+            for (String uriToken : testUri.split("/")) {
+                if (uriToken.startsWith(":")) {
+                    testRegex = testRegex.concat("/[\\w\\W]*");
+                    wildcardIndices.put(index, uriToken.substring(1));
+                } else if (!uriToken.equals("")) {
+                    testRegex = testRegex.concat("/" + uriToken);
                 }
-                testRegex += "$";
-
-                // If we have a match, parse the URI params and call that handler
-                if (uri.matches(testRegex)) {
-                    // TODO: Move this to a separate method 'parseUriParams'
-                    String[] uriTokens = uri.split("/");
-                    for (Map.Entry<Integer, String> wildcardIndexEntry : wildcardIndices.entrySet()) {
-                        int wildcardIndex = wildcardIndexEntry.getKey();
-                        uriParams.put(wildcardIndexEntry.getValue(), uriTokens[wildcardIndex]);
-                    }
-                    handler = routerMap.get(method).get(entry.getKey());
-                    break;
+                index++;
+            }
+            testRegex += "$";
+
+            // If we have a match, parse the URI params and call that handler
+            if (uri.matches(testRegex)) {
+                // TODO: Move this to a separate method 'parseUriParams'
+                String[] uriTokens = uri.split("/");
+                for (Map.Entry<Integer, String> wildcardIndexEntry : wildcardIndices.entrySet()) {
+                    int wildcardIndex = wildcardIndexEntry.getKey();
+                    params.put(wildcardIndexEntry.getValue(), uriTokens[wildcardIndex]);
                 }
+                handler = routerMap.get(method).get(entry.getKey());
+                break;
             }
+        }
 
-            return handler.handle(session, uriParams);
-        } catch (Exception e) {
-            // TODO: Don't show internal error messages in production, only show them in dev
-            return new InternalErrorResponse(e.getMessage());
+        BaseResponse res = handler.handle(session, params);
+        System.out.println("Finished processing " + method + " request for '" + uri + "'");
```

Comment:
There's an issue for that: https://github.com/appium/appium-espresso-driver/issues/11